src/qibosoq/programs/sweepers.py

```diff
@@ -0,0 +1,146 @@
+                std_register = self.get_gen_reg(gen_ch, sweep_type)
+                swept_register = self.new_gen_reg(gen_ch, name=f"sweep_bias_{gen_ch}")
+                self.bias_sweep_registers[gen_ch] = (swept_register, std_register)
+
+                max_gain = int(self.soccfg["gens"][gen_ch]["maxv"])
+                starts = (sweeper.starts * max_gain).astype(int)
+                stops = (sweeper.stops * max_gain).astype(int)
+
+                new_sweep = QickSweep(
+                    self,
+                    swept_register,  # sweeper_register
+                    starts[idx],  # start
+                    stops[idx],  # stop
+                    sweeper.expts,  # number of points
+                )
+                sweep_list.append(new_sweep)
+        else:
+            for idx, jdx in enumerate(sweeper.indexes):
+                pulse = self.sequence[jdx]
+                gen_ch = pulse.dac
+
+                sweep_type = SWEEPERS_TYPE[sweeper.parameters[idx]]
+                register = self.get_gen_reg(gen_ch, sweep_type)
+
+                if sweeper.parameters[idx] is Parameter.AMPLITUDE:
+                    max_gain = int(self.soccfg["gens"][gen_ch]["maxv"])
+                    starts = (sweeper.starts * max_gain).astype(int)
+                    stops = (sweeper.stops * max_gain).astype(int)
+
+                new_sweep = QickSweep(
+                    self,
+                    register,  # sweeper_register
+                    starts[idx],  # start
+                    stops[idx],  # stop
+                    sweeper.expts,  # number of points
+                )
+                sweep_list.append(new_sweep)
+
+        self.add_sweep(merge_sweeps(sweep_list))
+
+    def initialize(self):
+        """Declre nyquist zones for all the DACs and all the readout frequencies.
+
+        Function called by AveragerProgram.__init__.
+        """
+        self.declare_nqz_zones([pulse for pulse in self.sequence if pulse.type == "drive"])
+        self.declare_nqz_flux()
+        if self.is_mux:
+            self.declare_gen_mux_ro()
+        else:
+            self.declare_nqz_zones([pulse for pulse in self.sequence if pulse.type == "readout"])
+        self.declare_readout_freq()
+
+        self.pulses_registered = True
+        for pulse in self.sequence:
+            if self.is_mux:
+                if pulse.type != "drive":
+                    continue
+            self.add_pulse_to_register(pulse)
+
+        for sweeper in self.sweepers:
+            self.add_sweep_info(sweeper)
+
+        for _, registers in self.bias_sweep_registers.items():
+            swept_reg, non_swept_reg = registers
+            non_swept_reg.set_to(swept_reg)
+
+        self.sync_all(self.wait_initialize)
+
+
+SWEEPERS_TYPE = {
+    Parameter.FREQUENCY: "freq",
+    Parameter.AMPLITUDE: "gain",
+    Parameter.BIAS: "gain",
+    Parameter.RELATIVE_PHASE: "phase",
+    Parameter.START: "t",
+}
```

Comment:
This you could do it directly with the enum:

```py
class Parameter(Enum):
    """Available parameters for sweepers."""

    FREQUENCY = "freq"
    AMPLITUDE = "gain"
    ...
```
and use it like:
```py
Parameter[sweeper.parameters[0]].value
```

(unless you're already using somewhere that the enum should be a `IntEnum`, but it doesn't seem the case in `main`)

Reply:
Simple Enum was giving problem in the serialization (something similar to https://stackoverflow.com/questions/24481852/serialising-an-enum-member-to-json)...

Reply:
I see, that makes sense. An `IntEnum` is an enum whose variants are also `int`s, so it shouldn't be a problem to serialize.

For the great good and added benefits, in py3.11 they introduced [`StrEnum`](https://docs.python.org/3/library/enum.html#enum.StrEnum).
However, we're poor men living in a py3.9 world, so we should give up on the added benefits and just do the mixing ourselves:
```py
class Parameter(str, Enum):
    """Available parameters for sweepers."""

    FREQUENCY = "freq"
    AMPLITUDE = "gain"
    ...
```
I tested it serializes well with `json.dumps`, but I leave you the pleasure to test it in Qibosoq as well :)

Reply:
Nice!